mautic/mautic
# Preview variations for emails and landing pages

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



As a user I would like to be able to more easily preview emails & landing pages and their variations including:

Translations (if available)
A/B tests
Dynamic content & tokens - [ ]

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create email with at least one AB test and one translation
3. Use at leas in one template tokens like {contactfield=firstname}‍ {contactfield=lastname}‍ to see if contact info is filled properly
4. Test translations and variants preview in email detail page :
![106509198-96711a00-64cd-11eb-8d1c-d648c434fe82](https://user-images.githubusercontent.com/54406786/202433733-cafd5ed5-db68-40a4-8d5b-7b2d3fb52d0b.png)
5. Use the same scenario for pages

#### Steps to test permissions
Preview actions has now protected access for admins and other roles. Now it is not accessible by anonymous users. There's need to test úit under other role than admin which has no defined resource access right. I've `Sales` role for this.
1. Use admin account and preview s are accessible
2. Use `Sales` or other role defined in mautic with `Email permissions - View Own & View Others` and `Landing page permissions - View Own & View Others` - works with respect to this settings agains user with this role
3. Try to  access previews in anonymous window - this will return access denied
